app/celery/lookup_recipient_communication_permissions_task.py

```diff
@@ -116,3 +111,13 @@ def recipient_has_given_permission(
 
         # return status reason message if message should not be sent
         return "No recipient opt-in found for explicit preference"
+    else:
+        current_app.logger.info('Value of permission for item %s for recipient %s for notification %s: %s',
+                                communication_item.va_profile_item_id if communication_item else None,
+                                id_value, notification_id, is_allowed)
+
+        if is_allowed:
+            return None
+
+        # return status reason message if message should not be sent
+        return "Contact preferences set to false"
```

Comment:
As revised, now you don't need an else block because either "except" block will exit the code.  You only get here if no exception occurred.  Also note that you no longer need to test `if communication_item` because you already returned above if that value was None.

Finally, I would consolidate the return value:
```
return None if is_allowed else "Contact preferences set to false"
```